hr_leave_custom_hour_interval/models/hr_leave.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2021 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class HrLeave(models.Model):
+    _inherit = "hr.leave"
+
+    request_time_hour_from = fields.Float("Float hour from")
+
+    request_hour_from = fields.Char(compute="_compute_hour_from", store=True)
+
+    request_time_hour_to = fields.Float("Float hour to")
+
+    request_hour_to = fields.Char(compute="_compute_hour_to", store=True)
+
+    @api.depends("request_time_hour_from")
+    def _compute_hour_from(self):
+        for leave in self:
+            leave.request_hour_from = "%.2f" % self.request_time_hour_from
+
+    @api.depends("request_time_hour_to")
+    def _compute_hour_to(self):
+        for leave in self:
+            leave.request_hour_to = "%.2f" % self.request_time_hour_to
+
+    @api.depends("request_time_hour_from", "request_time_hour_to")
+    def _compute_date_from_to(self):
+        return super()._compute_date_from_to()
+
+    def action_validate(self):
```

Comment:
@xaviedoanhduy could you please share your ideas about this override?

`exclude_public_holidays` is not existed btw

Reply:
hi @dzungtran89, I am trying to fix failed [tests](https://github.com/OCA/hr-holidays/blob/2601f2c9a71d3efaa174a0993aec3cf7462f7c27/hr_holidays_public/tests/test_holidays_calculation.py#L188) in the `hr_holidays_public` module when overriding the action_validate function. 
some pr is exists do the same in [here](https://github.com/OCA/hr-holidays/pull/116)

Reply:
you can get the error log in [here](https://github.com/OCA/hr-holidays/actions/runs/9464172923/job/26070919033#step:8:137)

Reply:
``` Traceback (most recent call last):
  File "/__w/hr-holidays/hr-holidays/hr_holidays_public/tests/test_holidays_calculation.py", line 188, in test_number_of_hours_excluding_employee_2
    leave_request.action_validate()
  File "/__w/hr-holidays/hr-holidays/hr_leave_custom_hour_interval/models/hr_leave.py", line 47, in action_validate
    return super().action_validate()
  File "/__w/hr-holidays/hr-holidays/hr_holidays_public/models/hr_leave.py", line 22, in action_validate
    return super().action_validate()
  File "/opt/odoo/addons/hr_holidays/models/hr_leave.py", line 1385, in action_validate
    employee_requests._validate_leave_request()
  File "/opt/odoo/addons/hr_holidays/models/hr_leave.py", line 1112, in _validate_leave_request
    holiday.message_post(
  File "/opt/odoo/addons/mail/models/mail_thread.py", line 2096, in message_post
    raise ValueError(_("Posting a message should be done on a business document. Use message_notify to send a notification to an user."))
ValueError: Posting a message should be done on a business document. Use message_notify to send a notification to an user.
```

You could take some times to investigate the issue on the module `hr_holidays_public`, the module `hr_leave_custom_hour_interval` is not relevant

btw, A new commit is quite recent [odoo/odoo/commit/9f1aaec55fff6](https://github.com/odoo/odoo/commit/9f1aaec55fff6) which triggers a message post in the action_validate

Reply:
hi @dzungtran89,  the overide action_validate fuction is removed from this PR. so now I can create PR with OCA right?

Reply:
yes, please go ahead